### PR TITLE
Feature/48120 Migrate tab

### PR DESCRIPTION
**Pull request by jszczech**

### CODE REVIEW

File: components/02-elements/tab/Tab.vue [hunk 1]
```diff
@@ -0,0 +1,234 @@
+<template>
```
Comment (Igloczek) [inline components/02-elements/tab/Tab.vue]: Tabs not Tab

File: components/02-elements/tab/Tab.vue [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+      <button
+        :id="tab.tabId"
+        :key="'tab' + tab.tabId"
+        :class="[
+          'tab__title',
+          tab.tabId === activeTab && 'tab__title--active'
+        ]"
+        :data-tab="tab.tabId"
+        :aria-controls="tab.tabId"
+        :aria-selected="tab.tabId === activeTab ? 'true' : 'false'"
```
Comment (Igloczek) [inline components/02-elements/tab/Tab.vue]: IMO you can reduce it to
```suggestion
        :aria-selected="tab.tabId === activeTab"
```

File: components/02-elements/tab/Tab.vue [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  >
+    <template
+      v-for="tab in tabs"
+    >
+      <button
+        :id="tab.tabId"
+        :key="'tab' + tab.tabId"
+        :class="[
+          'tab__title',
+          tab.tabId === activeTab && 'tab__title--active'
```
Comment (Igloczek) [inline components/02-elements/tab/Tab.vue]: It looks cleaner this way
```suggestion
          { 'tab__title--active': tab.tabId === activeTab }
```

File: components/02-elements/tab/Tab.vue [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+      <div
+        v-if="tab.tabId === activeTab"
+        :id="tab.tabId"
+        :key="'tab__content' + tab.tabId"
+        :data-content="tab.tabId"
+        :aria-labelledby="tab.titleId"
+        :aria-hidden="tab.tabId === activeTab ? 'true' : 'false'"
+        :class="[
+          'tab__content',
+          tab.tabId === activeTab && 'tab__content--active'
```
Comment (Igloczek) [inline components/02-elements/tab/Tab.vue]:
```suggestion
          { 'tab__content--active': tab.tabId === activeTab }
```

File: components/02-elements/tab/Tab.vue [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+          :icon="icon"
+          :class="[
+            iconClass,
+            'button__icon'
+          ]"
+        />
+      </button>
+      <div
+        v-if="tab.tabId === activeTab"
+        :id="tab.tabId"
```
Comment (Igloczek) [inline components/02-elements/tab/Tab.vue]: Something is wrong here, you use the same ID for the button.

File: components/02-elements/tab/Tab.vue [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+    props: {
+      tabs: {
+        type: Array,
+        required: true
+      },
+      icon: {
+        type: String,
+        default: null
+      },
+      modifier: {
```
Comment (Igloczek) [inline components/02-elements/tab/Tab.vue]: What's the point of this prop?

File: components/02-elements/tab/Tab.vue [hunk 7]
```diff
@@ -0,0 +1,234 @@
+<template>
+  <div
+    :class="['tab', modifier && 'tab--' + modifier]"
+  >
+    <template
+      v-for="tab in tabs"
+    >
+      <button
+        :id="tab.tabId"
```
Comment (Igloczek) [inline components/02-elements/tab/Tab.vue]: These IDs should be autogenerated, there is no point in defining it manually.

Comment (Igloczek) [reply components/02-elements/tab/Tab.vue]: This one probably should be 
```suggestion
        :id="tab.titleId"
```

File: components/02-elements/tabs/Tabs.stories.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  data: () => ({
+    tabs
+  }),
+  template: `
+    <app>
+      <alpaca-tabs :tabs="tabs.tabs"/>
+    </app>
+  `
+}))
+.add('With icon', () => ({
```
Comment (Igloczek) [inline components/02-elements/tabs/Tabs.stories.js]: I don't see any diffrence between [default](https://alpaca-storybook-priqnl9by.now.sh/?selectedKind=Elements%2FTabs&selectedStory=Default&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel) and [with icon](https://alpaca-storybook-priqnl9by.now.sh/?selectedKind=Elements%2FTabs&selectedStory=With%20icon&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Fstories%2Fstories-panel)

File: components/02-elements/tabs/Tabs.vue [hunk 9]
```diff
@@ -0,0 +1,233 @@
+<template>
+  <div
+    class="tab"
+  >
+    <template
+      v-for="{ id, tab } in getTabsWithId"
+    >
```
Comment (szafran89) [inline components/02-elements/tabs/Tabs.vue]: IMO until we have just one attribute we can keep it in one line

File: components/02-elements/tabs/Tabs.vue [hunk 10]
```diff
@@ -0,0 +1,233 @@
+<template>
+  <div
+    class="tab"
+  >
```
Comment (szafran89) [inline components/02-elements/tabs/Tabs.vue]: IMO until we have just one attribute we can keep it in one line

File: components/01-globals/icon/Icon.vue [hunk 11]
```diff
@@ -1,14 +1,7 @@
 <template>
-  <svg
-    :class="[
-      'icon',
-      customClass
-    ]"
-  >
+  <svg class="icon">
```
Comment (szafran89) [inline components/01-globals/icon/Icon.vue]: missing `role` attr

File: components/02-elements/tabs/Tabs.stories.js [hunk 12]
```diff
@@ -0,0 +1,32 @@
...
+  components: { App, AlpacaTabs },
+  data: () => ({
+    tabs
+  }),
+  template: `
+    <app>
+      <alpaca-tabs 
+        :tabs="tabs.tabs"
```
Comment (szafran89) [inline components/02-elements/tabs/Tabs.stories.js]: maybe `tabs.items` seems better name?

File: components/02-elements/tabs/Tabs.vue [hunk 13]
```diff
@@ -0,0 +1,233 @@
...
+        :key="'tab__content' + tab.tabId"
+        :data-content="tab.tabId"
+        :aria-labelledby="tab.titleId"
+        :aria-hidden="tab.tabId === activeTab"
+        :class="[
+          'tab__content',
+          { 'tab__content--active': tab.tabId === activeTab }
+        ]"
+      >
+        {{ tab.content }}
```
Comment (szafran89) [inline components/02-elements/tabs/Tabs.vue]: What do you think to use the slot for tab content? Sometimes there are some reviews, ratings, tables and so on, not just plain text.

Comment (jszczech) [reply components/02-elements/tabs/Tabs.vue]: Now we iterate inside the component, so slot is not possible to use. First, we need rewrite this component in a different way. Should I do it?

File: components/02-elements/tabs/mocks/tabs.json [hunk 14]
```diff
@@ -0,0 +1,39 @@
...
+      "title": "Tab title",
+      "titleId": "tab-title-1",
+      "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque pharetra ut magna ornare lacinia. Cras sodales elit ac pellentesque aliquet. Nulla nec viverra turpis. Mauris eget quam interdum, viverra enim eget, ultricies purus. Suspendisse eleifend, turpis id pretium consectetur, massa nunc suscipit elit, a porta neque neque sed ex. Curabitur nec ante non urna rhoncus semper. Maecenas id pulvinar erat. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Quisque mollis sit amet quam id vestibulum. Nunc faucibus quam non venenatis laoreet. Pellentesque eu feugiat tellus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos.",
+      "active": true
+    },
+    {
+      "tabId": "tab-2",
+      "title": "Tab title 2",
+      "titleId": "tab-title-2",
+      "contentElement": "image"
```
Comment (szafran89) [inline components/02-elements/tabs/mocks/tabs.json]: For what is it?

File: components/02-elements/tabs/mocks/tabs.json [hunk 15]
```diff
@@ -0,0 +1,39 @@
+{
+  "icon": {
+    "iconId": "angle-down",
+    "iconTitle": "Arrow Down"
+  },
```
Comment (szafran89) [inline components/02-elements/tabs/mocks/tabs.json]: I think we don't need icon in mock file

File: components/02-elements/tabs/mocks/tabs.json (hunk 14 above)
Comment (szafran89) [reply components/02-elements/tabs/mocks/tabs.json]: I think we can remove it

**Pull request merged**